pkg/kubelet/kubelet_pods.go

```diff
@@ -607,7 +607,7 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 
 // podFieldSelectorRuntimeValue returns the runtime value of the given
 // selector for a pod.
-func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod *v1.Pod, podIP string) (string, error) {
+func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod *v1.Pod, podIP, hostIP string) (string, error) {
```

Comment:
We can get hostIP by invoking [GetHostIP](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_getters.go#L218) in this function, then it's no need to pass it in from run container options. Do I miss something?
BTW, as is commented in a dup PR https://github.com/kubernetes/kubernetes/pull/43666#discussion_r108087504 .